plugins/repository-hdfs/hadoop-client-api/build.gradle

```diff
@@ -1,16 +1,42 @@
-apply plugin: 'elasticsearch.build'
-apply plugin: 'com.gradleup.shadow'
+apply plugin: 'elasticsearch.java'
 
-dependencies {
-  implementation "org.apache.hadoop:hadoop-client-api:${project.parent.versions.hadoop}"
+sourceSets {
+  patcher
 }
 
-tasks.named('shadowJar').configure {
-  exclude 'org/apache/hadoop/util/ShutdownHookManager$*.class'
+configurations {
+  thejar {
+    canBeResolved = true
+  }
 }
 
-['jarHell', 'thirdPartyAudit', 'forbiddenApisMain', 'splitPackagesAudit'].each {
-  tasks.named(it).configure {
-    enabled = false
+dependencies {
+  thejar("org.apache.hadoop:hadoop-client-api:${project.parent.versions.hadoop}") {
+    transitive = false
   }
+
+  patcherImplementation 'org.ow2.asm:asm:9.7.1'
+  patcherImplementation 'org.ow2.asm:asm-tree:9.7.1'
+}
+
+def outputDir = layout.buildDirectory.dir("patched-classes")
+
+def patchTask = tasks.register("patchClasses", JavaExec)
+patchTask.configure {
+  dependsOn configurations.thejar, sourceSets.patcher.getCompileJavaTaskName()
+  outputs.dir(outputDir)
```

Comment:
We define outputs but no inputs here. A `dependsOn` doesn't imply an input, but an input _does_ imply a dependency.

Sounds like `configurations.thejar` and `sourcesets.patcher.output` should be inputs to this task.